### PR TITLE
remove types setuptools and release 1.31

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Please summarise the changes.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by cqc-melf** [comment]
Tries to solve https://github.com/CQCL/tket/issues/1513

File: pytket/docs/changelog.rst [hunk 1]
```diff
@@ -1,9 +1,10 @@
 Changelog
 =========
 
-(unreleased)
-------------
+1.31.0 (August 2024)
+--------------------
 
+* remove types-pkg_resources dependency
```
Comment (CalMacCQ) [inline pytket/docs/changelog.rst:7]: GIven that the removal of this package from pypi renders pytket uninstallable I think we should announce this more prominently in the docs. What do you think?

Comment (CalMacCQ) [reply pytket/docs/changelog.rst:7]: I can do this in a separate PR

**Comment by sjdilkes** [comment]
Looks like we might be in luck here - have you also tried installing in a fresh local environment where we can be sure there's no lingering installation of `types-pkg_resources`? I assume the CI is set up to do this, but a sanity check would be helful.

File: pytket/docs/changelog.rst (hunk 1 above)
Comment (sjdilkes) [inline pytket/docs/changelog.rst:7]:
```suggestion
* Remove unnecessary types-pkg_resources dependency due to package  being yanked from pypi
```

**Comment by jake-arkinstall** [comment]
A couple of thoughts of potential relevance:
* The version and release strings in pytket/docs/conf.py (I read the version from there when providing nix with the version information) still reference "1.30" - if this PR is for bumping the version should those be changed as part of it?
* poetry.lock still mentions pkg-resources

**Comment by cqc-melf** [comment]
> Looks like we might be in luck here - have you also tried installing in a fresh local environment where we can be sure there's no lingering installation of `types-pkg_resources`? I assume the CI is set up to do this, but a sanity check would be helful.

Yes I did that:
- [ ] install pytket
- [ ] run python tests
- [ ] run mypy checks
- [ ] make sure `types-pkg-resources` is still not installed

File: pytket/docs/changelog.rst (hunk 1 above)
Comment (cqc-melf) [reply pytket/docs/changelog.rst]: Yes we should do that. Feel free to make a second PR for that.

**Comment by cqc-melf** [comment]
The doc test run currently in an older version and I don't see and easy way to solve that at the moment. I would suggest removing the doc checks for now and try to add them back in later.

The main problem is:
Docs need poetry
Poetry does not like setup.py
(I think the solution should be building a wheel first and loading that device specific into poetry is a to complicated change for a release PR, and not fully sure that would work)

**Comment by cqc-melf** [comment]
@sjdilkes @CalMacCQ should be ready for review now. If there are any questions or anything we should discuss, happy to have a call.

**Review by CalMacCQ — APPROVED** [review]
Looks good to me.

The TKET webiste will pick up the v1.31 from pypi once the release is put out and will build the docs.

After the 1.31 release is out we can re-enable the docs test

**Review by sjdilkes — APPROVED** [review]

**Pull request merged**